pkg/controller/datavolume-controller.go

```diff
@@ -2823,14 +2823,27 @@ func (r *DatavolumeReconciler) detachPvcDeleteDv(pvc *corev1.PersistentVolumeCla
 }
 
 func updatePvcOwnerRefs(pvc *corev1.PersistentVolumeClaim, dv *cdiv1.DataVolume) {
-	refs := pvc.OwnerReferences
-	for i, r := range refs {
-		if r.UID == dv.UID {
-			pvc.OwnerReferences = append(refs[:i], refs[i+1:]...)
+	var refs []metav1.OwnerReference
+	for _, pvcOwnerRef := range pvc.OwnerReferences {
+		if pvcOwnerRef.UID != dv.UID {
+			refs = append(refs, pvcOwnerRef)
 			break
 		}
 	}
-	pvc.OwnerReferences = append(pvc.OwnerReferences, dv.OwnerReferences...)
+	for _, dvOwnerRef := range dv.OwnerReferences {
+		refExists := false
+		for _, pvcOwnerRef := range pvc.OwnerReferences {
+			if dvOwnerRef.UID == pvcOwnerRef.UID {
+				refExists = true
+				break
+			}
+		}
+		if !refExists {
+			dvOwnerRef.BlockOwnerDeletion = nil
```

Comment:
from https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/:
```
Kubernetes automatically sets this field to true if a [controller](https://kubernetes.io/docs/concepts/architecture/controller/) (for example, the Deployment controller) sets the value of the metadata.ownerReferences field.
```
I think we may be better with an explicit false value here, but maybe I got this wrong

Reply:
Sure, fixing.

Reply:
We should probably try to test this somehow that I am not lying :D
Maybe push the cdi-controller image to quay and we'll try to swap to it in a running cluster?

Reply:
awesome!